sdk/core/core/internal/az_span_internal.h

```diff
@@ -5,8 +5,11 @@
 #define _az_SPAN_INTERNAL_H
 
 #include <az_precondition_internal.h>
+#include <az_result.h>
 #include <az_span.h>
 
+#include <stdint.h>
```

Comment:
We shouldn't need this `#include`. What's requiring this?

Reply:
we use int32_t below

Reply:
I get that, we also use `int32_t` in `_az_span_diff`, so we didn't need it before. I think it is already being transitively included with `az_span.h` or some other header.

So, did you just add it as an abundance of  precaution, or was the build somehow failing now without it, when it should have been working fine before?

Reply:
Generally, we shouldn't rely on the headers we include to include the headers we expect. This is also the reason why we include our headers first, and then the standard library headers - so that we know that our header would compile even if customer did not include some stdlib header before including us.
If a function is declared in a header, and I am including that header, and implementing that function, then (within our SDK) I can expect that it did include the header to not only compile, but to use all the types that are input and output parameters for the function.
So, we use NULL in headers, we include stddef, bool - stdbool, `*int*_t` - stdint.